src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -171,27 +178,35 @@ async def infer_python_dependencies_via_imports(
     if not python_infer_subsystem.imports:
         return InferredDependencies([])
 
-    wrapped_tgt = await Get(WrappedTarget, Address, request.sources_field.address)
+    _wrapped_tgt = await Get(WrappedTarget, Address, request.sources_field.address)
+    tgt = _wrapped_tgt.target
     explicitly_provided_deps, parsed_imports = await MultiGet(
-        Get(ExplicitlyProvidedDependencies, DependenciesRequest(wrapped_tgt.target[Dependencies])),
+        Get(ExplicitlyProvidedDependencies, DependenciesRequest(tgt[Dependencies])),
         Get(
             ParsedPythonImports,
             ParsePythonImportsRequest(
                 cast(PythonSourceField, request.sources_field),
-                InterpreterConstraints.create_from_targets([wrapped_tgt.target], python_setup),
+                InterpreterConstraints.create_from_targets([tgt], python_setup),
                 string_imports=python_infer_subsystem.string_imports,
                 string_imports_min_dots=python_infer_subsystem.string_imports_min_dots,
             ),
         ),
     )
 
+    resolve = None
+    if tgt.has_field(PythonResolveField):
+        resolve_field = tgt[PythonResolveField]
+        resolve_field.validate(python_setup)
+        resolve = resolve_field.value_or_default(python_setup)
```

Comment:
I don't like how many lines this is taking to do. I'm thinking about ways to simplify this.